### PR TITLE
Return the value of the block for IO.popen

### LINKED ISSUE #8278: IO.popen does not return value of the provided block
According to the Ruby docs, if a block is provided to `IO.popen`, the value returned from the block becomes the value returned from the `popen` method.  In JRuby, this is never true and the IO object itself is always returned.  CRuby/MRI returns the value of the block as expected.

**Environment Information**

- jruby 9.4.7.0 (3.1.4) 2024-04-29 597ff08ac1 Java HotSpot(TM) 64-Bit Server VM 25.181-b13 on 1.8.0_181-b13 +jit [x86_64-mswin32]
- windows 10 enterprise

**Expected Behavior**

```
jruby-9.4.7.0\bin>.\jruby.exe -e "puts IO.popen('echo test')"
test
```

**Actual Behavior**

```
jruby-9.4.7.0\bin>.\jruby.exe -e "puts IO.popen('echo test')"
#<IO:0x1b11171f>
```

### PR DESCRIPTION
**Pull request by mullermp**
fixes #8278

Returns the value of the block for IO.popen and remove the test tag.

### CODE REVIEW

File: core/src/main/java/org/jruby/RubyIO.java [hunk 1]
```diff
@@ -4555,12 +4555,10 @@ public static IRubyObject popen(ThreadContext context, IRubyObject recv, IRubyOb
             io.setupPopen(runtime, modes, process);
 
             if (block.isGiven()) {
-                ensureYieldClose(context, io, block);
-
                 // RubyStatus uses real native status now, so we unshift Java's shifted exit status
                 context.setLastExitStatus(RubyProcess.RubyStatus.newProcessStatus(runtime, process.waitFor() << 8, ShellLauncher.getPidFromProcess(process)));
             }
-            return io;
+            return ensureYieldClose(context, io, block);
```
[the remark this answers is not in the document]
Comment (mullermp) [reply core/src/main/java/org/jruby/RubyIO.java:4561]: I was just following this: https://github.com/jruby/jruby/issues/8278#issuecomment-2174609944

[the remark this answers is not in the document]
Comment (mullermp) [reply core/src/main/java/org/jruby/RubyIO.java]: I'm not entirely sure I followed your comment correctly, but I pushed a change to this, feel free to change it again if not.

[the remark this answers is not in the document]
Comment (enebo) [reply core/src/main/java/org/jruby/RubyIO.java]: @mullermp Thanks.  That was exactly what I was looking for.

**Comment by mullermp** [comment]
I'm not sure why some of the tests failed. Can you handle this part?

**Comment by enebo** [comment]
@mullermp sure thing.  This does not seem related to either of your PRs and both of your PRs were green before merging (and this last change should not be the cause here).  If it is I will fix it up.  Thanks for your contribution!

**Pull request merged**